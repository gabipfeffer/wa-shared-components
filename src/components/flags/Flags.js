import React from 'react';
import cs from 'classnames';

import flagStyles from './flagStyles.pcss';
import s from './Flags.pcss';

type Props = {
  flagName: string,
  column: boolean,
  hideText: boolean,
  withTitle?: string
};

function Flags({ flagName, column, withTitle, hideText }: Props) {
  if (withTitle) {
    return (
      <div>
        <h2>
          {withTitle} <i className={cs(s.icon, flagStyles[`icon-flag-${flagName}`])} />
        </h2>
      </div>
    );
  }
  return (
    <span
      className={cs(s.container, { [s.column]: column })}
      style={hideText ? { width: 'auto' } : {}}>
      {hideText ? null : <span>{flagName}</span>}
      <i className={cs(s.icon, flagStyles[`icon-flag-${flagName}`])} />
    </span>
  );
}

Flags.defaultProps = {
  flagName: '',
  column: false,
  hideText: false
};

export default Flags;
