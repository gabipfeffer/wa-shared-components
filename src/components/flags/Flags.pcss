.container {
  display: flex;
  align-items: center;
  width: 60px;
  vertical-align: middle;

  i {
    margin-left: 4px;
  }

  span {
    display: inline-block;
    width: 28px;
  }

  .title {
    font-size: 1.5rem;
  }

  &.column {
    width: 24px;
    text-align: center;

    span {
      font-size: 13px;
      margin-bottom: 3px;
    }

    i {
      margin: 0 0 0 3px;
    }
  }
}

.icon {
  display: inline-block;
}
