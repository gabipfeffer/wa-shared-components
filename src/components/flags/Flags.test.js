import React from 'react';
import Flags from './Flags';

describe('Flags component', () => {
  it('render correctly with default props', () => {
    const wrapper = shallow(<Flags />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render correctly when column is false', () => {
    const wrapper = shallow(<Flags flagName={'AFG'} column={false}>AFG</Flags>);
  });
  it('is created with FLAG AFG and column true', () => {
    const wrapper = shallow(
      <Flags flagName={'AFG'} column>AFG</Flags>
    );
    expect(wrapper.find('span').at(1).text()).toBe('AFG');
  });
});
