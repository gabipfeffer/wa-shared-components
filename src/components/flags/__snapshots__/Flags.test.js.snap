// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags component render correctly with default props 1`] = `
<span
  className="container"
>
  <span />
  <i
    className="icon icon-flag-"
  />
</span>
`;
